### PR TITLE
Make it a composer project so it can, you know, work...

### PR DESCRIPTION
**Pull request by shauno**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/vendor/
```
Comment (Ed-leRoux) [inline .gitignore:1]: vendor*

File: src/ErrorCatalog.php [hunk 2]
```diff
@@ -5,21 +5,26 @@
...
+    protected $language;
     protected $catalog = [];
 
-    public function __construct($namespace)
+    /**
+     * ErrorCatalog constructor.
+     * @param $namespace
+     * @param string $language
+     */
+    public function __construct($namespace, $language = 'en-US')
```
Comment (Ed-leRoux) [inline src/ErrorCatalog.php]: `$this->language = $language`

**Review by Ed-leRoux — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (shauno) [reply .gitignore:1]: That's how the `composer init` command sets it, but ¯\\\_(ツ)\_/¯

Comment (shauno) [reply .gitignore:1]: Then should the `idea/` directory be the same: `.idea*`?
I think I prefer how it is now, you're trying to exclude the directory, you don't need a wildcard

File: src/ErrorCatalog.php [hunk 3]
```diff
@@ -4,22 +4,52 @@
...
         return $this;
     }
-}
\ No newline at end of file
+
+    /**
+     * @param $name
+     * @return ErrorCatalogItem
+     */
+    public function getItem($name)
```
Comment (shauno) [inline src/ErrorCatalog.php]: What should this method return if it doesn't find the item?

File: src/ErrorSpecIssue.php [hunk 4]
```diff
@@ -8,14 +8,36 @@ class ErrorSpecIssue
      * @var string $id The ID for the issue.
      */
     protected $id;
+
+    /**
+     * @var string The JSON pointer to identify the issue
+     */
+    protected $jsonPointer;
```
Comment (shauno) [inline src/ErrorSpecIssue.php]: It's not really a json pointer though is it? What exactly is this extra identifier?

File: .gitignore (hunk 1 above)
Comment (Ed-leRoux) [reply .gitignore:1]: 👍

File: src/ErrorSpecIssue.php (hunk 4 above)
Comment (Ed-leRoux) [reply src/ErrorSpecIssue.php]: Maybe call this `reference`

File: src/ErrorCatalog.php (hunk 3 above)
Comment (Ed-leRoux) [reply src/ErrorCatalog.php]: for now, `return null;`. Maybe we should add a default at a later stage

**Review by Ed-leRoux — CHANGES_REQUESTED** [review]

**Review by Ed-leRoux — APPROVED** [review]
👍

**Pull request merged**